components/add-layers/wmts/add-layers-wmts.component.js

```diff
@@ -70,93 +94,185 @@ export default {
-      } else {
-        $scope.currentLayer = layer;
-        const wmtsLayerPanel = HsLayoutService.contentWrapper.querySelector(
-          '.hs-wmts-layerpanel'
-        );
-        const layerNode = HsLayoutService.contentWrapper.querySelector(
-          '#wmtslayer-' + index
-        );
-        if (wmtsLayerPanel) {
-          layerNode.parentNode.insertBefore(
-            wmtsLayerPanel,
-            layerNode.nextSibling
-          );
+    $scope.checked = function () {
+      for (const layer of $scope.services) {
+        if (layer.checked) {
+          return true;
         }
       }
+      return false;
+    };
+    $scope.changed = function () {
+      $scope.isChecked = $scope.checked();
     };
 
     /**
      * @function addLayer
+     * @param checked
      * @memberOf hs.addLayersWmts.controller
      * @description Add layer to map
      * @param {object} layer - Wrapped layer to add
      */
+    $scope.tryAddLayers = function (checked) {
+      $scope.add_all = checked;
+      $scope.addLayers(checked);
+      return;
+    };
 
-    $scope.addLayer = function (layer) {
-      const projection = get($scope.map_projection);
-      const projectionExtent = projection.getExtent();
-      for (let idx = 0; idx < $scope.services.TileMatrixSet.length; idx++) {
-        if (
-          $scope.services.TileMatrixSet[idx].Identifier == $scope.tileMatrixSet
-        ) {
-          $scope.layerTileMatrix = $scope.services.TileMatrixSet[idx];
+    $scope.addLayers = function (checked) {
+      /**
+       * @param layer
+       */
+      function recurse(layer) {
+        if (!checked || layer.checked) {
+          $scope.addLayer(layer);
         }
-      }
-      const size =
-        getWidth(projectionExtent) /
-        $scope.layerTileMatrix.TileMatrix[0].TileWidth;
-      const resolutions = new Array($scope.layerTileMatrix.TileMatrix.length);
-      const matrixIds = new Array($scope.layerTileMatrix.TileMatrix.length);
-      for (let z = 0; z < $scope.layerTileMatrix.TileMatrix.length; ++z) {
-        // generate resolutions and matrixIds arrays for this WMTS
-        resolutions[z] = size / Math.pow(2, z);
-        matrixIds[z] = z;
-      }
-
-      const dimensions = {};
 
-      angular.forEach(layer.Dimension, (val) => {
-        dimensions[val.name] = val;
+        angular.forEach(layer.Layer, (sublayer) => {
+          recurse(sublayer);
+        });
+      }
+      angular.forEach($scope.services, (layer) => {
+        recurse(layer);
       });
+      HsLayoutService.setMainPanel('layermanager');
+    };
 
-      const new_layer = new Tile({
-        title: layer.Title,
-        source: new WMTS({
-          url: $scope.tileURL,
+    /**
+     * Returns prefered tile format 
+     *
+     * @memberof hs.addLayersWMTS
+     * @function getPreferedFormat
+     * @param {object} formats Set of avaliable formats for layer being added
+     */
+    $scope.getPreferedFormat = function (formats) {
```

Comment:
```suggestion
    $scope.getPreferredFormat = function (formats) {
```

and analogously to all mis-spelled `prefered`